### PR TITLE
ETL Refactor 

### PR DESCRIPTION
**Pull request by pomadchin**
EMR demo based on spark-etl

- [x] implemented custom InputPlugin, to load landsat data
- [x] refactor Makefile
- [x] add json job definition
- [x] test on some bbox and time interval
- [x] merge master
- [x] multibackend support
- [x] move to cassandra 3

```bash
make local-ingest # to run tests ingest using tests json conf files
make local-tile-server # to start server
```

### CODE REVIEW

File: ingest/src/main/scala/demo/etl/landsat/LandsatInput.scala [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  val name = "landsat"
+
+  val bandsWanted = Array(
+    // Red, Green, Blue
+    "4", "3", "2",
+    // Near IR
+    "5",
+    "QA")
+
+  var images: Seq[LandsatImage] = Seq()
```
Comment (pomadchin) [inline ingest/src/main/scala/demo/etl/landsat/LandsatInput.scala]: smth really bad, used that in order not to load same data twice

File: ingest/src/main/scala/demo/LandsatIngest.scala [hunk 2]
```diff
@@ -129,40 +63,27 @@ object LandsatIngestMain extends Logging {
...
 
-    val bandsWanted = Array(
-      // Red, Green, Blue
-      "4", "3", "2",
-      // Near IR
-      "5",
-      "QA")
+    EtlConf(args).getEtlJobs.foreach { job =>
+      val etl = Etl(job, Etl.defaultModules :+ LandsatModule)
+      val inputPlugin = new TemporalMultibandLandsatInput()
```
Comment (pomadchin) [inline ingest/src/main/scala/demo/LandsatIngest.scala]: However, we need explicitly create input plugin class there, to have access to `var images` field, and not to download lc8 scenes twice

**Comment by pomadchin** [comment]
Don't really like the way we need to work with etl to define smth custom, think further refactor steps required. https://github.com/geotrellis/geotrellis/pull/1553

File: server/src/main/scala/demo/Render.scala [hunk 3]
```diff
@@ -31,7 +31,12 @@ object Render {
       val red = tile.band(0).convert(IntCellType).map(clamp _).normalize(min, max, 0, 255)
       val green = tile.band(1).convert(IntCellType).map(clamp _).normalize(min, max, 0, 255)
       val blue = tile.band(2).convert(IntCellType).map(clamp _).normalize(min, max, 0, 255)
-      (red, green, blue)
+
+      println(red.asciiDraw())
+      //println(green.asciiDraw())
+      //println(blue.asciiDraw())
+
+      (red, red, red)
```
Comment (lossyrob) [inline server/src/main/scala/demo/Render.scala]: ?

Comment (pomadchin) [inline server/src/main/scala/demo/Render.scala]: oops / some debug code ._.

**Comment by pomadchin** [comment]
includes #6 and #16

**Comment by echeipesh** [comment]
@pomadchin still WIP ?

**Comment by pomadchin** [comment]
yep; would make final comments to describe problems / possible solutions / causes

File: project/Version.scala [hunk 4]
```diff
@@ -1,4 +1,4 @@
 object Version {
-  val geotrellis = "0.10.0"
+  val geotrellis = "1.0.0-SNAPSHOT"
```
Comment (echeipesh) [inline project/Version.scala]: needs to be SHA tag instead of `SNAPSHOT`

Comment (pomadchin) [inline project/Version.scala]: still depends on https://github.com/geotrellis/geotrellis/pull/1606

**Comment by echeipesh** [comment]
+1

**Pull request merged**